src/colorizejs.js

```diff
@@ -25,33 +25,48 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */
 
-$.fn.colorize = function(colors) {
-  var data = parseFloat(this.text());
-  var keys = Object.keys(colors)
-    .map(function (k) { return parseInt(k); })
-    .sort(function (a,b) { return a - b; })
-    .reverse();
-  var used = '';
-  for (i = 0; i < keys.length; ++i) {
-    var max = keys[i];
-    if (data >= max) {
-      var c = colors[max];
-      if (c.startsWith('.')) {
-        this.addClass(c.substring(1));
-      } else {
-        this.css('color', c);
-      }
-      used = c;
-      break;
-    }
-  }
-  for (i = 0; i < keys.length; ++i) {
-    var u = colors[keys[i]];
-    if (used != u) {
-      if (u.startsWith('.')) {
-        this.removeClass(u.substring(1));
-      }
-    }
-  }
-  return this;
+$.fn.colorize = function (colors) {
+	const data = parseFloat(this.text());
+	const sortedThresholds = getSortedThresholds(colors);
+	const appliedColor = findAndApplyColor(this, data, sortedThresholds, colors);
+	cleanupUnusedColors(this, sortedThresholds, colors, appliedColor);
+
+	return this;
 };
+
+function getSortedThresholds(colors) {
+	return Object.keys(colors)
+		.map((key) => parseInt(key))
+		.sort((a, b) => b - a);
+}
+
+function findAndApplyColor(element, data, thresholds, colors) {
+	for (let i = 0; i < thresholds.length; i++) {
+		const threshold = thresholds[i];
+
```

Comment:
@MaksZhukov take a look: https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

Reply:
@yegor256 Thanks, I've changed it.